compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -3540,8 +3562,24 @@ private boolean isAssignableToUnionType(BType source, BType target, Set<TypePair
                     break;
                 }
 
-                if (isAssignable(sourceMember, targetMember, unresolvedTypes)) {
-                    sourceTypeIsNotAssignableToAnyTargetType = false;
+                if (sourceMember.tag == TypeTags.ERROR) {
+                    if (targetUnionIsDistinctError) {
+                        sourceTypeIsNotAssignableToAnyTargetType = !isErrorTypeAssignable((BErrorType) sourceMember,
+                                (BErrorType) targetMember, unresolvedTypes);
+                    } else if (targetMember.tag == TypeTags.ERROR && allTargetTypesErrors) {
```

Comment:
its better to use
`} else if (allTargetTypesErrors && targetMember.tag == TypeTags.ERROR) {`

Reply:
Changed